Assets/MixedRealityToolkit/Interfaces/CameraSystem/IMixedRealityCameraSystem.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.MixedReality.Toolkit.CameraSystem
+{
+    public interface IMixedRealityCameraSystem : IMixedRealityEventSystem, IMixedRealityEventSource, IMixedRealityDataProvider
```

Comment:
I'm not sure we should be using an interface here. Nothing is calling IsOpaque via the interface and I don't think it sufficiently describes what a type must implement to be a camera system.

Reply:
It's thin at the moment but this is a baby step. The long term goal is to move ALL camera-related babysitting into this service.

If we don't want it to be this thin up front maybe we could expose profile-driven settings like camera background color, clear flags, etc?

Reply:
I'd say let's make it an interface when/if we actually use the camera system through its interface and if the interface defines well what is expected of a camera system. I.e. by implementing the interface I should have a reasonable chance of creating a type that can work well in place of MixedRealityCameraSystem.

Reply:
@luis-valverde-ms it's obvious you don't understand how the services work.

It's required to define an interface so to set it up in the configuration profile.

This PR effectively uses the camera service via interface.

Reply:
@luis-valverde-ms I agree with what you're saying but our hands are tied in the sense that an interface is required for this to be a service. And if it's not a service then the main benefit of making this move - the ability to swap it out for something custom - is lost.

Here's one idea. This PR leaves out two of the biggest camera-related behaviors in the toolkit - finding/creating the main camera and play space transform. Right now they're handled in CameraCache.cs and MixedRealityToolkit.cs. My plan was always to move those responsibilities into the camera service, but I left them out to keep the initial PR small.

If we move those right away, the interface would look (roughly) like this:

```
public interface IMixedRealityCameraSystem 
{
    // Supplies a main camera, either by finding or creating it.
    public Camera MainCamera { get; }

    // Supplies the parent transform for the main camera. If you want to move the player around, use this transform.
    public Transform PlayspaceTransform { get; }
}
```